packages/carbon-web-components/src/components/combo-box/combo-box.ts

```diff
@@ -17,6 +17,8 @@ import CDSDropdown, { DROPDOWN_KEYBOARD_ACTION } from '../dropdown/dropdown';
 import CDSComboBoxItem from './combo-box-item';
 import styles from './combo-box.scss';
 import { carbonElement as customElement } from '../../globals/decorators/carbon-element';
+import { ifDefined } from 'lit/directives/if-defined';
+import ifNonEmpty from '../../globals/directives/if-non-empty';
```

Comment:
I think we can slowly start removing `ifNonEmpty`, since the latest implementation of [ifDefined](https://lit.dev/docs/templates/directives/#ifdefined) covers both `null` and `undefined`? In fact, after double-checking, `ifNonEmpty` is using `ifDefined` under the hood.

Reply:
Hum, I changed it, but then I reverted the change. Here I actually want `ifNonEmpty` since, where it's used, it references a string variable that defaults to the empty string, and I want to leave the attribute out if its set to the empty string. 

There is a difference between the two. `ifNonEmpty` will avoid outputting the attribute if the value is the empty string, in addition to when its `null` or `undefined`. See [this lit playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQgeyBodG1sLCBjc3MsIExpdEVsZW1lbnQgfSBmcm9tIFwibGl0XCI7XG5pbXBvcnQgeyBjdXN0b21FbGVtZW50LCBwcm9wZXJ0eSB9IGZyb20gXCJsaXQvZGVjb3JhdG9ycy5qc1wiO1xuaW1wb3J0IHsgaWZEZWZpbmVkIH0gZnJvbSBcImxpdC9kaXJlY3RpdmVzL2lmLWRlZmluZWQuanNcIjtcblxuY29uc3QgaWZOb25FbXB0eSA9ICh2YWx1ZSkgPT5cbiAgaWZEZWZpbmVkKHZhbHVlID09PSBcIlwiID8gdW5kZWZpbmVkIDogdmFsdWUgPz8gdW5kZWZpbmVkKTtcblxuQGN1c3RvbUVsZW1lbnQoXCJzaW1wbGUtZ3JlZXRpbmdcIilcbmV4cG9ydCBjbGFzcyBTaW1wbGVHcmVldGluZyBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgIHAge1xuICAgICAgY29sb3I6IGJsdWU7XG4gICAgfVxuICBgO1xuXG4gIEBwcm9wZXJ0eSgpXG4gIG5hbWUgPSBcIlNvbWVib2R5XCI7XG5cbiAgcHJpdmF0ZSBmb28gPSBcIlwiO1xuXG4gIHByaXZhdGUgYmFyID0gXCJiYXJcIjtcblxuICBwcml2YXRlIGJhejtcblxuICByZW5kZXIoKSB7XG4gICAgY29uc3QgeyBmb28sIGJhciwgYmF6IH0gPSB0aGlzO1xuICAgIHJldHVybiBodG1sYDxwXG4gICAgICBkYXRhLWZvby1pZmRlZmluZWQ9XCIke2lmRGVmaW5lZChmb28pfVwiXG4gICAgICBkYXRhLWJhci1pZmRlZmluZWQ9XCIke2lmRGVmaW5lZChiYXIpfVwiXG4gICAgICBkYXRhLWJhei1pZmRlZmluZWQ9XCIke2lmRGVmaW5lZChiYXopfVwiXG4gICAgICBkYXRhLWZvby1ub25lbXB0eT1cIiR7aWZOb25FbXB0eShmb28pfVwiXG4gICAgICBkYXRhLWJhci1ub25lbXB0eT1cIiR7aWZOb25FbXB0eShiYXIpfVwiXG4gICAgICBkYXRhLWJhei1ub25lbXB0eT1cIiR7aWZOb25FbXB0eShiYXopfVwiXG4gICAgPlxuICAgICAgSGVsbG8sICR7dGhpcy5uYW1lfSFcbiAgICA8L3A-YDtcbiAgfVxufVxuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxzaW1wbGUtZ3JlZXRpbmcgbmFtZT1cIldvcmxkXCI-PC9zaW1wbGUtZ3JlZXRpbmc-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMy4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4yLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeNC4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjMuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d).

Reply:
Huh that's interesting. I could swear I read in Lit v3 docs that `ifDefined` now includes the behavior of `iFnonEmpty`. Disregard then, my bad!